### ISSUE TITLE
[Remote-SSH Bug]: Cannot connect (MacOS ARM to Raspberry Pi)

### ISSUE THREAD

**Issue by freesurfer-rge** [microsoft/vscode-remote-release#10557]
### Is there an existing issue for this bug?

- [x] I have searched the existing issues

### Required Troubleshooting Steps

- [x] I have followed these troubleshooting steps
- [x] I have tried both values of the `remote.SSH.useLocalServer` setting [(more info here)](https://github.com/microsoft/vscode-remote-release/wiki/Remote-SSH-troubleshooting#troubleshooting-steps)

### Connect Locally

It doesn’t connect successfully (different error message)

### ->

With useLocalServer set to True, the generated command is `ssh -v -T -D 51229 -o ConnectTimeout=15 pi02`

This appears to connect, but doesn't actually give me a shell prompt (not sure if this is expected):

```bash
(base) rge21@Mac ~ % ssh -v -T -D 51229 -o ConnectTimeout=15 pi02
OpenSSH_9.8p1, LibreSSL 3.3.6
debug1: Reading configuration data /Users/rge21/.ssh/config
debug1: /Users/rge21/.ssh/config line 1: Applying options for pi02
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: Reading configuration data /etc/ssh/ssh_config.d/100-macos.conf
debug1: /etc/ssh/ssh_config.d/100-macos.conf line 1: Applying options for *
debug1: Reading configuration data /etc/ssh/crypto.conf
debug1: /etc/ssh/ssh_config line 57: Applying options for *
debug1: Authenticator provider $SSH_SK_PROVIDER did not resolve; disabling
debug1: Connecting to pi02.local [fe80::a141:a6ff:8a68:215b%en1] port 22.
debug1: fd 5 clearing O_NONBLOCK
debug1: Connection established.
debug1: identity file /Users/rge21/.ssh/freesurfer.rge.pi02 type 3
debug1: identity file /Users/rge21/.ssh/freesurfer.rge.pi02-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_9.8
debug1: Remote protocol version 2.0, remote software version OpenSSH_7.9p1 Raspbian-10+deb10u4
debug1: compat_banner: match: OpenSSH_7.9p1 Raspbian-10+deb10u4 pat OpenSSH* compat 0x04000000
debug1: Authenticating to pi02.local:22 as 'pi'
debug1: load_hostkeys: fopen /Users/rge21/.ssh/known_hosts2: No such file or directory
debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts: No such file or directory
debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts2: No such file or directory
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: algorithm: ecdh-sha2-nistp256
debug1: kex: host key algorithm: ssh-ed25519
debug1: kex: server->client cipher: aes128-gcm@openssh.com MAC: <implicit> compression: none
debug1: kex: client->server cipher: aes128-gcm@openssh.com MAC: <implicit> compression: none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: SSH2_MSG_KEX_ECDH_REPLY received
debug1: Server host key: ssh-ed25519 SHA256:pVmzdHllnIgrqb2p7Omn7LDsjnrW0M9oW846IY7W8dM
debug1: load_hostkeys: fopen /Users/rge21/.ssh/known_hosts2: No such file or directory
debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts: No such file or directory
debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts2: No such file or directory
debug1: Host 'pi02.local' is known and matches the ED25519 host key.
debug1: Found key in /Users/rge21/.ssh/known_hosts:3
debug1: ssh_packet_send2_wrapped: resetting send seqnr 3
debug1: rekey out after 4294967296 blocks
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: ssh_packet_read_poll2: resetting read seqnr 3
debug1: SSH2_MSG_NEWKEYS received
debug1: rekey in after 4294967296 blocks
debug1: SSH2_MSG_EXT_INFO received
debug1: kex_ext_info_client_parse: server-sig-algs=<ssh-ed25519,ssh-rsa,rsa-sha2-256,rsa-sha2-512,ssh-dss,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521>
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey,password
debug1: Next authentication method: publickey
debug1: get_agent_identities: bound agent to hostkey
debug1: get_agent_identities: agent returned 1 keys
debug1: Will attempt key: /Users/rge21/.ssh/freesurfer.rge.pi02 ED25519 SHA256:NVGA+QG89kDyNheadjl19kiRfYTrd1PokI+lpx+wQvk explicit agent
debug1: Offering public key: /Users/rge21/.ssh/freesurfer.rge.pi02 ED25519 SHA256:NVGA+QG89kDyNheadjl19kiRfYTrd1PokI+lpx+wQvk explicit agent
debug1: Server accepts key: /Users/rge21/.ssh/freesurfer.rge.pi02 ED25519 SHA256:NVGA+QG89kDyNheadjl19kiRfYTrd1PokI+lpx+wQvk explicit agent
Authenticated to pi02.local ([fe80::a141:a6ff:8a68:215b%en1]:22) using "publickey".
debug1: Local connections to LOCALHOST:51229 forwarded to remote address socks:0
debug1: Local forwarding listening on ::1 port 51229.
debug1: channel 0: new port-listener [port listener] (inactive timeout: 0)
debug1: Local forwarding listening on 127.0.0.1 port 51229.
debug1: channel 1: new port-listener [port listener] (inactive timeout: 0)
debug1: channel 2: new session [client-session] (inactive timeout: 0)
debug1: Requesting no-more-sessions@openssh.com
debug1: Entering interactive session.
debug1: pledge: exec
debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0
debug1: client_input_hostkeys: searching /Users/rge21/.ssh/known_hosts for pi02.local / (none)
debug1: client_input_hostkeys: searching /Users/rge21/.ssh/known_hosts2 for pi02.local / (none)
debug1: client_input_hostkeys: hostkeys file /Users/rge21/.ssh/known_hosts2 does not exist
debug1: client_input_hostkeys: host key found matching a different name/address, skipping UserKnownHostsFile update
debug1: Remote: /home/pi/.ssh/authorized_keys:1: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
debug1: Remote: /home/pi/.ssh/authorized_keys:1: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
debug1: Requesting X11 forwarding with authentication spoofing.
debug1: Requesting authentication agent forwarding.
debug1: Sending environment.
debug1: channel 2: setting env LANG = "en_GB.UTF-8"
Linux pi02 5.10.103+ #1529 Tue Mar 8 12:19:18 GMT 2022 armv6l

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
```

With useLocalServer set to False, the command generated is `ssh -T -D 51262 -o ConnectTimeout=15 pi02 sh` which doesn't produce any output. If I throw in a `-v` then I get:

```bash
OpenSSH_9.8p1, LibreSSL 3.3.6
debug1: Reading configuration data /Users/rge21/.ssh/config
debug1: /Users/rge21/.ssh/config line 1: Applying options for pi02
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: Reading configuration data /etc/ssh/ssh_config.d/100-macos.conf
debug1: /etc/ssh/ssh_config.d/100-macos.conf line 1: Applying options for *
debug1: Reading configuration data /etc/ssh/crypto.conf
debug1: /etc/ssh/ssh_config line 57: Applying options for *
debug1: Authenticator provider $SSH_SK_PROVIDER did not resolve; disabling
debug1: Connecting to pi02.local [fe80::a141:a6ff:8a68:215b%en1] port 22.
debug1: fd 5 clearing O_NONBLOCK
debug1: Connection established.
debug1: identity file /Users/rge21/.ssh/freesurfer.rge.pi02 type 3
debug1: identity file /Users/rge21/.ssh/freesurfer.rge.pi02-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_9.8
debug1: Remote protocol version 2.0, remote software version OpenSSH_7.9p1 Raspbian-10+deb10u4
debug1: compat_banner: match: OpenSSH_7.9p1 Raspbian-10+deb10u4 pat OpenSSH* compat 0x04000000
debug1: Authenticating to pi02.local:22 as 'pi'
debug1: load_hostkeys: fopen /Users/rge21/.ssh/known_hosts2: No such file or directory
debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts: No such file or directory
debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts2: No such file or directory
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: algorithm: ecdh-sha2-nistp256
debug1: kex: host key algorithm: ssh-ed25519
debug1: kex: server->client cipher: aes128-gcm@openssh.com MAC: <implicit> compression: none
debug1: kex: client->server cipher: aes128-gcm@openssh.com MAC: <implicit> compression: none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: SSH2_MSG_KEX_ECDH_REPLY received
debug1: Server host key: ssh-ed25519 SHA256:pVmzdHllnIgrqb2p7Omn7LDsjnrW0M9oW846IY7W8dM
debug1: load_hostkeys: fopen /Users/rge21/.ssh/known_hosts2: No such file or directory
debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts: No such file or directory
debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts2: No such file or directory
debug1: Host 'pi02.local' is known and matches the ED25519 host key.
debug1: Found key in /Users/rge21/.ssh/known_hosts:3
debug1: ssh_packet_send2_wrapped: resetting send seqnr 3
debug1: rekey out after 4294967296 blocks
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: ssh_packet_read_poll2: resetting read seqnr 3
debug1: SSH2_MSG_NEWKEYS received
debug1: rekey in after 4294967296 blocks
debug1: SSH2_MSG_EXT_INFO received
debug1: kex_ext_info_client_parse: server-sig-algs=<ssh-ed25519,ssh-rsa,rsa-sha2-256,rsa-sha2-512,ssh-dss,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521>
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey,password
debug1: Next authentication method: publickey
debug1: get_agent_identities: bound agent to hostkey
debug1: get_agent_identities: agent returned 1 keys
debug1: Will attempt key: /Users/rge21/.ssh/freesurfer.rge.pi02 ED25519 SHA256:NVGA+QG89kDyNheadjl19kiRfYTrd1PokI+lpx+wQvk explicit agent
debug1: Offering public key: /Users/rge21/.ssh/freesurfer.rge.pi02 ED25519 SHA256:NVGA+QG89kDyNheadjl19kiRfYTrd1PokI+lpx+wQvk explicit agent
debug1: Server accepts key: /Users/rge21/.ssh/freesurfer.rge.pi02 ED25519 SHA256:NVGA+QG89kDyNheadjl19kiRfYTrd1PokI+lpx+wQvk explicit agent
Authenticated to pi02.local ([fe80::a141:a6ff:8a68:215b%en1]:22) using "publickey".
debug1: Local connections to LOCALHOST:51262 forwarded to remote address socks:0
debug1: Local forwarding listening on ::1 port 51262.
debug1: channel 0: new port-listener [port listener] (inactive timeout: 0)
debug1: Local forwarding listening on 127.0.0.1 port 51262.
debug1: channel 1: new port-listener [port listener] (inactive timeout: 0)
debug1: channel 2: new session [client-session] (inactive timeout: 0)
debug1: Requesting no-more-sessions@openssh.com
debug1: Entering interactive session.
debug1: pledge: exec
debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0
debug1: client_input_hostkeys: searching /Users/rge21/.ssh/known_hosts for pi02.local / (none)
debug1: client_input_hostkeys: searching /Users/rge21/.ssh/known_hosts2 for pi02.local / (none)
debug1: client_input_hostkeys: hostkeys file /Users/rge21/.ssh/known_hosts2 does not exist
debug1: client_input_hostkeys: host key found matching a different name/address, skipping UserKnownHostsFile update
debug1: Remote: /home/pi/.ssh/authorized_keys:1: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
debug1: Remote: /home/pi/.ssh/authorized_keys:1: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
debug1: Requesting X11 forwarding with authentication spoofing.
debug1: Requesting authentication agent forwarding.
debug1: Sending environment.
debug1: channel 2: setting env LANG = "en_GB.UTF-8"
debug1: Sending command: sh
```
so it does seem to be connecting at least.

### Expected Behavior

That the connection works? If I issue `ssh pi02` in any terminal, the connection is immediate.

### Steps To Reproduce

Running on a fresh install of VSCode on a new Sequoia install on MacOS ARM. Connection is being made to a Raspberry Pi Zero.

As a side bug, can I also add, that I usually have to cajole VS Code into showing my the `Remote-SSH` option in the Output window. When the connection fails, it vanishes, which is *not the most useful behaviour*.

### Remote-SSH Log

<details>
<summary>Remote-SSH Log</summary>
<p>

```
[20:43:21.316] Log Level: 2
[20:43:21.323] SSH Resolver called for "ssh-remote+pi02", attempt 1
[20:43:21.324] remote.SSH.useLocalServer = true
[20:43:21.324] remote.SSH.useExecServer = true
[20:43:21.324] remote.SSH.path = 
[20:43:21.324] remote.SSH.configFile = 
[20:43:21.324] remote.SSH.useFlock = true
[20:43:21.324] remote.SSH.lockfilesInTmp = false
[20:43:21.324] remote.SSH.localServerDownload = auto
[20:43:21.324] remote.SSH.remoteServerListenOnSocket = false
[20:43:21.324] remote.SSH.defaultExtensions = []
[20:43:21.324] remote.SSH.loglevel = 2
[20:43:21.324] remote.SSH.enableDynamicForwarding = true
[20:43:21.324] remote.SSH.enableRemoteCommand = false
[20:43:21.324] remote.SSH.serverPickPortsFromRange = {}
[20:43:21.324] remote.SSH.serverInstallPath = {}
[20:43:21.324] remote.SSH.permitPtyAllocation = false
[20:43:21.324] remote.SSH.preferredLocalPortRange = undefined
[20:43:21.324] remote.SSH.useCurlAndWgetConfigurationFiles = false
[20:43:21.324] remote.SSH.experimental.chat = false
[20:43:21.324] remote.SSH.experimental.enhancedSessionLogs = false
[20:43:21.327] VS Code version: 1.96.0
[20:43:21.327] Remote-SSH version: remote-ssh@0.116.0
[20:43:21.327] darwin arm64
[20:43:21.328] SSH Resolver called for host: pi02
[20:43:21.328] Setting up SSH remote "pi02"
[20:43:21.331] Acquiring local install lock: /var/folders/8b/2h5rbpmj5_j61g1gdztzgs400000gp/T/vscode-remote-ssh-49bb6a3c-install.lock
[20:43:21.333] Looking for existing server data file at /Users/rge21/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-49bb6a3c-138f619c86f1199955d53b4166bef66ef252935c-0.116.0-es/data.json
[20:43:21.333] No existing data file
[20:43:21.333] Using commit id "138f619c86f1199955d53b4166bef66ef252935c" and quality "stable" for server
[20:43:21.334] Script variables:
 {
  "ExitCode.0": "OK",
  "ExitCode.64": "LinuxPrereqs",
  "ExitCode.65": "ProxyAuthenticationRequired",
  "ExitCode.128": "UnparsableOutput",
  "ExitCode.129": "EmptyOutput",
  "ExitCode.131": "UnhandledResolverExitCode",
  "ExitCode.132": "OfflineError",
  "ExitCode.133": "SshInstall",
  "ExitCode.134": "FindLocalPort",
  "ExitCode.135": "Timeout",
  "ExitCode.136": "ExhaustedRetries",
  "ExitCode.137": "PermissionDenied",
  "ExitCode.138": "MitmPortForwardingDisabled",
  "ExitCode.139": "Canceled",
  "ExitCode.140": "BadLocalDownloadRequest",
  "ExitCode.141": "LocalDownloadFailed",
  "ExitCode.142": "LocalLockTimeout",
  "ExitCode.143": "NoBash",
  "ExitCode.144": "InvalidHostName",
  "ExitCode.145": "BadInstallScriptResult",
  "ExitCode.146": "ShellRequestFailedOnChannel",
  "ExitCode.147": "InvalidPortRange",
  "ExitCode.148": "SendDisconnectBrokenPipe",
  "ExitCode.149": "LocalServerCrashed",
  "ExitCode.150": "NoFreePortInRange",
  "ExitCode.151": "InvalidCustomServerPath",
  "ExitCode.192": "AlreadyInProgress",
  "ExitCode.193": "ServerDownloadFailed",
  "ExitCode.194": "NoDownloaderAvailable",
  "ExitCode.195": "NoDownloaderAvailableForStatusCheck",
  "ExitCode.196": "UnsupportedArch",
  "ExitCode.197": "StatusCheckFailed",
  "ExitCode.198": "NeedInsidersArch",
  "ExitCode.199": "ServerTransferFailed",
  "ExitCode.200": "ServerFailedToStart",
  "ExitCode.201": "NeedInsidersWindows",
  "ExitCode.202": "CreateInstallDirFailed",
  "ExitCode.203": "UnsupportedPlatform",
  "ExitCode.204": "ServerTerminatedCVE20201416",
  "ExitCode.205": "UnpackFailed",
  "ExitCode.206": "ChangeDirFailed",
  "ExitCode.1023": "UnclassifiedError",
  "ExitCode.OK": "0",
  "ExitCode.LinuxPrereqs": "64",
  "ExitCode.ProxyAuthenticationRequired": "65",
  "ExitCode.UnparsableOutput": "128",
  "ExitCode.EmptyOutput": "129",
  "ExitCode.UnhandledResolverExitCode": "131",
  "ExitCode.OfflineError": "132",
  "ExitCode.SshInstall": "133",
  "ExitCode.FindLocalPort": "134",
  "ExitCode.Timeout": "135",
  "ExitCode.ExhaustedRetries": "136",
  "ExitCode.PermissionDenied": "137",
  "ExitCode.MitmPortForwardingDisabled": "138",
  "ExitCode.Canceled": "139",
  "ExitCode.BadLocalDownloadRequest": "140",
  "ExitCode.LocalDownloadFailed": "141",
  "ExitCode.LocalLockTimeout": "142",
  "ExitCode.NoBash": "143",
  "ExitCode.InvalidHostName": "144",
  "ExitCode.BadInstallScriptResult": "145",
  "ExitCode.ShellRequestFailedOnChannel": "146",
  "ExitCode.InvalidPortRange": "147",
  "ExitCode.SendDisconnectBrokenPipe": "148",
  "ExitCode.LocalServerCrashed": "149",
  "ExitCode.NoFreePortInRange": "150",
  "ExitCode.InvalidCustomServerPath": "151",
  "ExitCode.AlreadyInProgress": "192",
  "ExitCode.ServerDownloadFailed": "193",
  "ExitCode.NoDownloaderAvailable": "194",
  "ExitCode.NoDownloaderAvailableForStatusCheck": "195",
  "ExitCode.UnsupportedArch": "196",
  "ExitCode.StatusCheckFailed": "197",
  "ExitCode.NeedInsidersArch": "198",
  "ExitCode.ServerTransferFailed": "199",
  "ExitCode.ServerFailedToStart": "200",
  "ExitCode.NeedInsidersWindows": "201",
  "ExitCode.CreateInstallDirFailed": "202",
  "ExitCode.UnsupportedPlatform": "203",
  "ExitCode.ServerTerminatedCVE20201416": "204",
  "ExitCode.UnpackFailed": "205",
  "ExitCode.ChangeDirFailed": "206",
  "ExitCode.UnclassifiedError": "1023",
  "InstallUnpackCode.Success": "success",
  "InstallUnpackCode.Error": "error",
  "InstallUnpackCode.MissingFiles": "missingFiles",
  "uuid": "1caf30186635",
  "startMarker": "1caf30186635: running",
  "commitId": "138f619c86f1199955d53b4166bef66ef252935c",
  "quality": "stable",
  "token": "11111111-a1a1-11a1-11a1-aa111a1a11aa",
  "vscodeAgentFolder": "$HOME/.vscode-server",
  "allowClientDownload": "1",
  "forceClientDownload": "0",
  "cliNameInArchive": "code",
  "ignoreWgetConfigFlag": " --no-config ",
  "ignoreCurlConfigFlag": " --disable ",
  "wgetTriesSegment": "--tries=1",
  "listenArgs": "--on-host=127.0.0.1 --on-port",
  "getDownloadServerStartTrigger": "1caf30186635:trigger_server_download",
  "getDownloadServerEndTrigger": "1caf30186635:trigger_server_download_end",
  "getProgressDownloading": "1caf30186635%%1%%",
  "getProgressInstalling": "1caf30186635%%2%%",
  "printVars": "echo \"SSH_AUTH_SOCK==$SSH_AUTH_SOCK==\"\necho \"DISPLAY==$DISPLAY==\""
}
[20:43:21.336] Install and start server if needed
[20:43:21.339] PATH: /Users/rge21/miniconda3/bin:/Users/rge21/miniconda3/condabin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin:/Volumes/External 1M2 01/Extra Applications/Visual Studio Code.app/Contents/Resources/app/bin
[20:43:21.339] Checking ssh with "ssh -V"
[20:43:21.345] > OpenSSH_9.8p1, LibreSSL 3.3.6

[20:43:21.347] askpass server listening on /var/folders/8b/2h5rbpmj5_j61g1gdztzgs400000gp/T/vscode-ssh-askpass-5a20905c9f1af7146bac1357c3444cd909a9bb81.sock
[20:43:21.347] Spawning local server with {"serverId":1,"ipcHandlePath":"/var/folders/8b/2h5rbpmj5_j61g1gdztzgs400000gp/T/vscode-ssh-askpass-a2d9ff5b7dca27bca16d16866192db217f92660f.sock","sshCommand":"ssh","sshArgs":["-v","-T","-D","51280","-o","ConnectTimeout=15","pi02"],"serverDataFolderName":".vscode-server","dataFilePath":"/Users/rge21/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-49bb6a3c-138f619c86f1199955d53b4166bef66ef252935c-0.116.0-es/data.json"}
[20:43:21.347] Local server env: {"SSH_AUTH_SOCK":"/private/tmp/com.apple.launchd.VJkrJtaVVX/Listeners","SHELL":"/bin/zsh","DISPLAY":"/private/tmp/com.apple.launchd.Z75kjKcLau/org.xquartz:0","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"/Users/rge21/.vscode/extensions/ms-vscode-remote.remote-ssh-0.116.0/out/local-server/askpass.sh","VSCODE_SSH_ASKPASS_NODE":"/Volumes/External 1M2 01/Extra Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Plugin).app/Contents/MacOS/Code Helper (Plugin)","VSCODE_SSH_ASKPASS_EXTRA_ARGS":"","VSCODE_SSH_ASKPASS_MAIN":"/Users/rge21/.vscode/extensions/ms-vscode-remote.remote-ssh-0.116.0/out/askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"/var/folders/8b/2h5rbpmj5_j61g1gdztzgs400000gp/T/vscode-ssh-askpass-5a20905c9f1af7146bac1357c3444cd909a9bb81.sock"}
[20:43:21.348] Spawned 6220
[20:43:21.348] Using connect timeout of 17 seconds
[20:43:21.401] > local-server-1> Running ssh connection command: ssh -v -T -D 51280 -o ConnectTimeout=15 pi02
[20:43:21.402] > local-server-1> Spawned ssh, pid=6225
[20:43:21.407] stderr> OpenSSH_9.8p1, LibreSSL 3.3.6
[20:43:21.408] stderr> ssh: connect to host pi02.local port 22: No route to host
[20:43:21.409] > local-server-1> ssh child died, shutting down
[20:43:21.411] Local server exit: 0
[20:43:21.411] Received install output: local-server-1> Running ssh connection command: ssh -v -T -D 51280 -o ConnectTimeout=15 pi02
local-server-1> Spawned ssh, pid=6225
OpenSSH_9.8p1, LibreSSL 3.3.6
ssh: connect to host pi02.local port 22: No route to host
local-server-1> ssh child died, shutting down

[20:43:21.411] WARN: $PLATFORM is undefined in installation script output.  Errors may be dropped.
[20:43:21.412] Resolver error: Error: No route to host
	at v.Offline (/Users/rge21/.vscode/extensions/ms-vscode-remote.remote-ssh-0.116.0/out/extension.js:2:696354)
	at /Users/rge21/.vscode/extensions/ms-vscode-remote.remote-ssh-0.116.0/out/extension.js:2:693068
	at t.handleInstallOutput (/Users/rge21/.vscode/extensions/ms-vscode-remote.remote-ssh-0.116.0/out/extension.js:2:693211)
	at e (/Users/rge21/.vscode/extensions/ms-vscode-remote.remote-ssh-0.116.0/out/extension.js:2:748331)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
	at async /Users/rge21/.vscode/extensions/ms-vscode-remote.remote-ssh-0.116.0/out/extension.js:2:770795
	at async t.withShowDetailsEvent (/Users/rge21/.vscode/extensions/ms-vscode-remote.remote-ssh-0.116.0/out/extension.js:2:774472)
	at async /Users/rge21/.vscode/extensions/ms-vscode-remote.remote-ssh-0.116.0/out/extension.js:2:745043
	at async P (/Users/rge21/.vscode/extensions/ms-vscode-remote.remote-ssh-0.116.0/out/extension.js:2:743098)
	at async t.resolveWithLocalServer (/Users/rge21/.vscode/extensions/ms-vscode-remote.remote-ssh-0.116.0/out/extension.js:2:744595)
	at async x (/Users/rge21/.vscode/extensions/ms-vscode-remote.remote-ssh-0.116.0/out/extension.js:2:767790)
	at async t.resolve (/Users/rge21/.vscode/extensions/ms-vscode-remote.remote-ssh-0.116.0/out/extension.js:2:771887)
	at async /Users/rge21/.vscode/extensions/ms-vscode-remote.remote-ssh-0.116.0/out/extension.js:2:1054592
[20:43:21.414] ------
```

</p>
</details>

### Anything else?

Above log is when useLocalServer is set to True.

Relevant section of `.ssh/config`

```
Host pi02
Hostname pi02.local
User  pi
ForwardX11 yes
ForwardX11Trusted yes
ForwardAgent yes
AddKeysToAgent yes
IdentityFile ~/.ssh/freesurfer.rge.pi02
```

**Comment by joshspicer** [comment]
Thanks for the detailed issue report. From the log this doesn't appear to be an issue with Remote - SSH.  The extension ran the ssh command `ssh -v -T -D 51280 -o ConnectTimeout=15 pi02`, and `ssh` failed with `ssh: connect to host pi02.local port 22: No route to host`.  I recognize that it's hard to parse these logs and the connection error was in the middle of the log.

What happens if you run that command manually outside of VS Code?  
 
```
[20:43:21.411] Received install output: local-server-1> Running ssh connection command: ssh -v -T -D 51280 -o ConnectTimeout=15 pi02
local-server-1> Spawned ssh, pid=6225
OpenSSH_9.8p1, LibreSSL 3.3.6
ssh: connect to host pi02.local port 22: No route to host
```

Also, I don't believe we can currently support Raspberry Pi zeros due to not having a VS Code Server that supports that architecture. See https://github.com/microsoft/vscode-remote-release/issues/669 and https://code.visualstudio.com/docs/remote/linux.   This would become more obvious in the logs if the SSH connection were succeeding.

**Comment by freesurfer-rge** [comment]
> What happens if you run that command manually outside of VS Code?

Isn't that what I showed in this block

```bash
OpenSSH_9.8p1, LibreSSL 3.3.6
debug1: Reading configuration data /Users/rge21/.ssh/config
debug1: /Users/rge21/.ssh/config line 1: Applying options for pi02
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: Reading configuration data /etc/ssh/ssh_config.d/100-macos.conf
debug1: /etc/ssh/ssh_config.d/100-macos.conf line 1: Applying options for *
debug1: Reading configuration data /etc/ssh/crypto.conf
debug1: /etc/ssh/ssh_config line 57: Applying options for *
debug1: Authenticator provider $SSH_SK_PROVIDER did not resolve; disabling
debug1: Connecting to pi02.local [fe80::a141:a6ff:8a68:215b%en1] port 22.
debug1: fd 5 clearing O_NONBLOCK
debug1: Connection established.
debug1: identity file /Users/rge21/.ssh/freesurfer.rge.pi02 type 3
debug1: identity file /Users/rge21/.ssh/freesurfer.rge.pi02-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_9.8
debug1: Remote protocol version 2.0, remote software version OpenSSH_7.9p1 Raspbian-10+deb10u4
debug1: compat_banner: match: OpenSSH_7.9p1 Raspbian-10+deb10u4 pat OpenSSH* compat 0x04000000
debug1: Authenticating to pi02.local:22 as 'pi'
debug1: load_hostkeys: fopen /Users/rge21/.ssh/known_hosts2: No such file or directory
debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts: No such file or directory
debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts2: No such file or directory
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: algorithm: ecdh-sha2-nistp256
debug1: kex: host key algorithm: ssh-ed25519
debug1: kex: server->client cipher: aes128-gcm@openssh.com MAC: <implicit> compression: none
debug1: kex: client->server cipher: aes128-gcm@openssh.com MAC: <implicit> compression: none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: SSH2_MSG_KEX_ECDH_REPLY received
debug1: Server host key: ssh-ed25519 SHA256:pVmzdHllnIgrqb2p7Omn7LDsjnrW0M9oW846IY7W8dM
debug1: load_hostkeys: fopen /Users/rge21/.ssh/known_hosts2: No such file or directory
debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts: No such file or directory
debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts2: No such file or directory
debug1: Host 'pi02.local' is known and matches the ED25519 host key.
debug1: Found key in /Users/rge21/.ssh/known_hosts:3
debug1: ssh_packet_send2_wrapped: resetting send seqnr 3
debug1: rekey out after 4294967296 blocks
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: ssh_packet_read_poll2: resetting read seqnr 3
debug1: SSH2_MSG_NEWKEYS received
debug1: rekey in after 4294967296 blocks
debug1: SSH2_MSG_EXT_INFO received
debug1: kex_ext_info_client_parse: server-sig-algs=<ssh-ed25519,ssh-rsa,rsa-sha2-256,rsa-sha2-512,ssh-dss,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521>
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey,password
debug1: Next authentication method: publickey
debug1: get_agent_identities: bound agent to hostkey
debug1: get_agent_identities: agent returned 1 keys
debug1: Will attempt key: /Users/rge21/.ssh/freesurfer.rge.pi02 ED25519 SHA256:NVGA+QG89kDyNheadjl19kiRfYTrd1PokI+lpx+wQvk explicit agent
debug1: Offering public key: /Users/rge21/.ssh/freesurfer.rge.pi02 ED25519 SHA256:NVGA+QG89kDyNheadjl19kiRfYTrd1PokI+lpx+wQvk explicit agent
debug1: Server accepts key: /Users/rge21/.ssh/freesurfer.rge.pi02 ED25519 SHA256:NVGA+QG89kDyNheadjl19kiRfYTrd1PokI+lpx+wQvk explicit agent
Authenticated to pi02.local ([fe80::a141:a6ff:8a68:215b%en1]:22) using "publickey".
debug1: Local connections to LOCALHOST:51262 forwarded to remote address socks:0
debug1: Local forwarding listening on ::1 port 51262.
debug1: channel 0: new port-listener [port listener] (inactive timeout: 0)
debug1: Local forwarding listening on 127.0.0.1 port 51262.
debug1: channel 1: new port-listener [port listener] (inactive timeout: 0)
debug1: channel 2: new session [client-session] (inactive timeout: 0)
debug1: Requesting no-more-sessions@openssh.com
debug1: Entering interactive session.
debug1: pledge: exec
debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0
debug1: client_input_hostkeys: searching /Users/rge21/.ssh/known_hosts for pi02.local / (none)
debug1: client_input_hostkeys: searching /Users/rge21/.ssh/known_hosts2 for pi02.local / (none)
debug1: client_input_hostkeys: hostkeys file /Users/rge21/.ssh/known_hosts2 does not exist
debug1: client_input_hostkeys: host key found matching a different name/address, skipping UserKnownHostsFile update
debug1: Remote: /home/pi/.ssh/authorized_keys:1: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
debug1: Remote: /home/pi/.ssh/authorized_keys:1: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
debug1: Requesting X11 forwarding with authentication spoofing.
debug1: Requesting authentication agent forwarding.
debug1: Sending environment.
debug1: channel 2: setting env LANG = "en_GB.UTF-8"
debug1: Sending command: sh
```

Which does contain the line `debug1: Connection established.` Did I miss something @joshspicer ?
